q2_diversity/tests/test_beta.py

```diff
@@ -456,5 +457,149 @@ def test_get_distance_boxplot_data_between_order_retained(self):
+        md = qiime2.MetadataCategory(
+            pd.Series([1, 2, 3, 4], name='number',
+                      index=['sample1', 'sample2', 'sample3', 'sample4']))
+
+        with tempfile.TemporaryDirectory() as output_dir:
+            beta_correlation(output_dir, dm, md)
+            index_fp = os.path.join(output_dir, 'index.html')
+            self.assertTrue(os.path.exists(index_fp))
+            # all expected plots are generated
+            self.assertTrue(os.path.exists(
+                            os.path.join(output_dir,
+                                         'beta-correlation-scatter.pdf')))
+            self.assertTrue(os.path.exists(
+                            os.path.join(output_dir,
+                                         'beta-correlation-scatter.png')))
+            self.assertTrue('Mantel test results' in open(index_fp).read())
+            self.assertTrue('r statistic' in open(index_fp).read())
+            self.assertTrue('Warning' in open(index_fp).read())
+
+    def test_error_on_missing_metadata(self):
+        dm = skbio.DistanceMatrix([[0.00, 0.25, 0.25],
+                                   [0.25, 0.00, 0.00],
+                                   [0.25, 0.00, 0.00]],
+                                  ids=['sample1', 'sample2', 'sample3'])
+        md = qiime2.MetadataCategory(
+            pd.Series([1, 2], name='number',
+                      index=['sample1', 'sample2']))
+
+        with tempfile.TemporaryDirectory() as output_dir:
+            with self.assertRaisesRegex(ValueError, 'no data: sample3'):
+                beta_correlation(output_dir, dm, md)
+
+    def test_error_on_nan_metadata(self):
+        dm = skbio.DistanceMatrix([[0.00, 0.25, 0.25],
+                                   [0.25, 0.00, 0.00],
+                                   [0.25, 0.00, 0.00]],
+                                  ids=['sample1', 'sample2', 'sample3'])
+        md = qiime2.MetadataCategory(
+            pd.Series([1.0, 2.0, ''], name='number',
+                      index=['sample1', 'sample2', 'sample3']))
+
+        with tempfile.TemporaryDirectory() as output_dir:
+            with self.assertRaisesRegex(ValueError, 'no data: sample3'):
+                beta_correlation(output_dir, dm, md)
+
+    def test_error_on_non_numeric_metadata(self):
+        dm = skbio.DistanceMatrix([[0.00, 0.25, 0.25],
+                                   [0.25, 0.00, 0.00],
+                                   [0.25, 0.00, 0.00]],
+                                  ids=['sample1', 'sample2', 'sample3'])
+        md = qiime2.MetadataCategory(
+            pd.Series([1.0, 2.0, 'hello-world'], name='number',
+                      index=['sample1', 'sample2', 'sample3']))
+
+        with tempfile.TemporaryDirectory() as output_dir:
+            with self.assertRaisesRegex(ValueError, 'Non-numeric data was'):
+                beta_correlation(output_dir, dm, md)
+
+    def test_basic_pearson(self):
```

Comment:
Can you add a similar test for passing a different number of permutations than 999?